File: C/Array/bubbleSort.c
#include<stdio.h>
/*
max element at last 

Computation time :
 T(n) = (n-1)n/2 = theta(n^2)
*/

void swap(int* a, int*b ){
    int temp = *a;
    *a = *b;
    *b = temp;
}
void printArr(int arr[],int n){
    for (int i = 0; i < n ; i++){
        printf("%d ",arr[i]);
    }
    printf("\n");
}
void sort(int arr[],int n){
    for(int i = 0; i < n-1 ; i++)
    {
        for (int j = 0; j < n-1-i; j++){
                if(arr[j] > arr[j+1]){
                    swap(&arr[j],&arr[j+1]);
                }
        }
    printArr(arr,n);
    }
}


int main(){
    int arr[] = { 8, 5,9, 2, 3, 1,7};
    printf("bubble sort\n");
    sort(arr,7);
    return 0;
}

File: Java/Array/Bubble_Sort.java
import java.util.Arrays;

class Main {
  static void bubbleSort(int array[]) {
    int size = array.length;
    // applying loop on every element
    for (int i = 0; i < size - 1; i++)
      // compare the elements of the array with a loop
      for (int j = 0; j < size - i - 1; j++)
        // compare each elements in array
        if (array[j] > array[j + 1]) {
          // swap elements if it's not in order
          int temp = array[j];
          array[j] = array[j + 1];
          array[j + 1] = temp;
        }
  }

  public static void main(String args[]) {
    int[] data = { 5, 3, 4, 1, 2 };
    Main.bubbleSort(data);
    
    System.out.println("Array sorted with bubble sort: ");
    System.out.println(Arrays.toString(data));
  }
}